examples/relative_hydration.py

```diff
@@ -0,0 +1,197 @@
+
+    gaff_filename = get_data_filename('data/gaff.xml')
+    system_generator = SystemGenerator([gaff_filename, 'amber99sbildn.xml', 'tip3p.xml'])
+    geometry_engine = FFAllAngleGeometryEngine()
+    proposal_engine = SmallMoleculeSetProposalEngine(
+        [initial_smiles, final_smiles], system_generator, residue_name=mol_name)
+
+    #generate topology proposal
+    topology_proposal = proposal_engine.propose(solvated_system, solvated_topology)
+
+    #generate new positions with geometry engine
+    new_positions, _ = geometry_engine.propose(topology_proposal, solvated_positions, beta)
+
+    return topology_proposal, solvated_positions, new_positions
+
+def generate_vacuum_hybrid_topology(mol_name="naphthalene", ref_mol_name="benzene"):
+    from topology_proposal import SmallMoleculeSetProposalEngine, TopologyProposal
+    import simtk.openmm.app as app
+    from openmoltools import forcefield_generators
+
+    from perses.tests.utils import createOEMolFromIUPAC, createSystemFromIUPAC, get_data_filename
+
+    m, unsolv_old_system, pos_old, top_old = createSystemFromIUPAC(mol_name)
+    refmol = createOEMolFromIUPAC(ref_mol_name)
+
+    initial_smiles = oechem.OEMolToSmiles(m)
+    final_smiles = oechem.OEMolToSmiles(refmol)
+
+    gaff_xml_filename = get_data_filename("data/gaff.xml")
+    forcefield = app.ForceField(gaff_xml_filename, 'tip3p.xml')
+    forcefield.registerTemplateGenerator(forcefield_generators.gaffTemplateGenerator)
+
+    solvated_system = forcefield.createSystem(top_old)
+
+    gaff_filename = get_data_filename('data/gaff.xml')
+    system_generator = SystemGenerator([gaff_filename, 'amber99sbildn.xml', 'tip3p.xml'])
+    geometry_engine = FFAllAngleGeometryEngine()
+    proposal_engine = SmallMoleculeSetProposalEngine(
+        [initial_smiles, final_smiles], system_generator, residue_name=mol_name)
+
+    #generate topology proposal
+    topology_proposal = proposal_engine.propose(solvated_system, top_old)
+
+    #generate new positions with geometry engine
+    new_positions, _ = geometry_engine.propose(topology_proposal, pos_old, beta)
+
+    return topology_proposal, pos_old, new_positions
+
+
+def check_alchemical_hybrid_elimination_bar(topology_proposal, old_positions, new_positions, ncmc_nsteps=50, n_iterations=50, NSIGMA_MAX=6.0, geometry=False):
+    """
+    Check that the hybrid topology, where both endpoints are identical, returns a free energy within NSIGMA_MAX of 0.
+    Parameters
+    ----------
+    topology_proposal
+    positions
+    ncmc_nsteps
+    NSIGMA_MAX
+
+    Returns
+    -------
+
+    """
+
+    #make the hybrid topology factory:
+    factory = HybridTopologyFactory(topology_proposal, old_positions, new_positions)
+
+    platform = openmm.Platform.getPlatformByName("CUDA")
+
+    hybrid_system = factory.hybrid_system
+    hybrid_topology = factory.hybrid_topology
+    initial_hybrid_positions = factory.hybrid_positions
+
+
+    #alchemical functions
+    functions = {
```

Comment:
I did. We should merge a different branch. It could be pretty annoying to reconcile all of those commits into this